### PR TITLE
Add schedule pagination to plugin API

### PR DESCRIPTION
**Pull request by matiasb**
Related to #1289

### CODE REVIEW

**Review by vadimkerr — APPROVED** [review]
Backend changes LGTM 👍 

Have you checked how this change works with `oncall_users` property on the view? It uses `get_queryset()`, will it take pagination into account?

**Comment by matiasb** [comment]
> Backend changes LGTM +1
> 
> Have you checked how this change works with `oncall_users` property on the view? It uses `get_queryset()`, will it take pagination into account?

That's a good point! Checking and pushing any related updates.

File: grafana-plugin/src/containers/EscalationVariants/parts/EscalationVariantsPopup.tsx [hunk 1]
```diff
@@ -155,10 +155,10 @@ const EscalationVariantsPopup = observer((props: EscalationVariantsPopupProps) =
             onChange={setSchedulesSearchTerm}
           />
           <GTable
-            emptyText={store.scheduleStore.getSearchResult() ? 'No schedules found' : 'Loading...'}
+            emptyText={store.scheduleStore.getSearchResult().results ? 'No schedules found' : 'Loading...'}
```
Comment (teodosii) [inline grafana-plugin/src/containers/EscalationVariants/parts/EscalationVariantsPopup.tsx:158]: Shouldn't this check for NPE? `getSearchResult` seems to return undefined for one case

File: grafana-plugin/src/containers/EscalationVariants/parts/EscalationVariantsPopup.tsx [hunk 2]
```diff
@@ -155,10 +155,10 @@ const EscalationVariantsPopup = observer((props: EscalationVariantsPopupProps) =
...
             rowKey="id"
             columns={scheduleColumns}
-            data={store.scheduleStore.getSearchResult()}
+            data={store.scheduleStore.getSearchResult().results}
```
Comment (teodosii) [inline grafana-plugin/src/containers/EscalationVariants/parts/EscalationVariantsPopup.tsx:161]: Same here, I guess we should check above the `GTable` for undefined

**Review by teodosii — APPROVED** [review]

**Review by maskin25 — APPROVED** [review]

**Pull request merged**